src/main/scala/is/hail/io/vcf/HtsjdkRecordReader.scala

```diff
@@ -178,18 +174,15 @@ case class GenotypeRecordReader(vcfSettings: VCFSettings) extends HtsjdkRecordRe
 
       val ad = g.getAD
       if (g.hasAD) {
-        if (vcfSettings.skipBadAD && ad.length != nAlleles)
-          reportAcc += VCFReport.ADInvalidNumber
-        else
+        if (!vcfSettings.skipBadAD || ad.length == nAlleles)
```

Comment:
I think this is clearer:

`if (ad.length == nAlleles || !vcfSettings.skipBadAD)`